puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy

```diff
@@ -0,0 +1,170 @@
+def ruby = '2.4'
+
+pipeline {
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup Git Repos') {
+            steps {
+                deleteDir()
+
+                sh "git config --global user.email 'ci@theforeman.org'"
+                sh "git config --global user.name 'jenkins'"
+
+                checkout changelog: true, poll: false, scm: [
+                    $class: 'GitSCM',
+                    branches: [[name: '${ghprbActualCommit}']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'PreBuildMerge', options: [fastForwardMode: 'FF', mergeRemote: 'origin', mergeStrategy: 'default', mergeTarget: '${ghprbTargetBranch}']]],
+                    userRemoteConfigs: [
+                        [refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*', url: 'https://github.com/katello/katello']
+                    ]
+                ]
+
+                dir('foreman') {
+                    git url: "https://github.com/theforeman/foreman", branch: "develop"
+                }
+            }
+        }
+        stage("Setup RVM") {
+            steps {
+
+                sh "echo test"
+
+                    withRVM(['rvm gemset empty --force'], ruby)
+                    withRVM(['gem update --no-ri --no-rdoc'], ruby)
+                    withRVM(['gem install bundler'], ruby)
+
+            }
+        }
+        stage('Configure Environment') {
+            steps {
+
+                dir('foreman') {
+                    sh 'mkdir -p config/settings.plugins.d'
+
+                    sh "cp config/settings.yaml.example config/settings.yaml"
+                    sh "sed -i 's/:locations_enabled: false/:locations_enabled: true/' config/settings.yaml"
+                    sh "sed -i 's/:organizations_enabled: false/:organizations_enabled: true/' config/settings.yaml"
+
+                    sh 'echo "gemspec :path => \'../\', :development_group => :katello_dev" >> bundler.d/Gemfile.local.rb'
+                    sh 'echo "gem \'psych\'" >> bundler.d/Gemfile.local.rb'
+
+                    sh "cp $HOME/postgresql.db.yaml config/database.yml"
+
+                    sh "sed -i \"s/database:.*/database: ${gemset()}-test/\" config/database.yml"
+                    sh """
+cat <<EOT >> config/database.yml
```

Comment:
I wounder if changing these from `sh` shells to use `dir`, `writeFile`, ect.. would have an impact on running time ?